lib/websocket.js

```diff
@@ -528,7 +551,12 @@ function initAsClient(websocket, address, protocols, options) {
     opts.auth = `${parsedUrl.username}:${parsedUrl.password}`;
   }
 
-  if (isUnixSocket) {
+  if (isWinPipe) {
+    const index = opts.path.indexOf(':', 11);
```

Comment:
Wouldn't `index` be incorrectly set to `-1` is `opts.path` is something like `'\\.\pipe\x:/foo'`, `'\\.\pipe\x'` is a valid named pipe.